src/models/deployers/PresaleTemplateDeployer.js

```diff
@@ -0,0 +1,279 @@
+
+    // new instance
+    const instanceReceipt = await this.presaleTemplate.newInstance(
+      instanceOwner,
+      id,
+      collateralToken,
+      bondedToken,
+      period,
+      exchangeRate,
+      openDate,
+      reserveRatio,
+      batchBlocks,
+      slippage
+    )
+
+    const Kernel = await this.environment.getArtifact('Kernel', '@aragon/os')
+    const installedApps = getInstalledAppsById(instanceReceipt, APP_IDS, Kernel)
+    Object.keys(installedApps).map((name) => {
+      const addresses = installedApps[name]
+      if (addresses.length == 0) {
+        delete installedApps[name]
+        return
+      }
+      logger.success(`Created ${name} contract at ${addresses[0]}`)
+      this._saveDeploy({ [name]: { address: addresses[0], transactionHash: instanceReceipt.tx }})
+    })
+
+    return installedApps
+  }
+
+  async _verifyFundraisingContracts(installedApps) {
+    for (const app in installedApps) {
+      await this._verifyProxy(app, installedApps[app][0])
+    }
+  }
+
+  _printPresaleInstanceDeploy(instanceOwner) {
+    logger.info('Deploying new Presale instance with config:')
+    logger.info(` - Owner:                                   ${instanceOwner}`)
+    logger.info(` - Id:                                      ${this.config.instance.id}`)
+    logger.info(` - Collateral token:                        ${this.config.instance.collateralToken}`)
+    logger.info(` - Bonded token:                            ${this.config.instance.bondedToken}`)
+    logger.info(` - Period:                                  ${this.config.instance.period}`)
+    logger.info(` - Exchange rate:                           ${this.config.instance.exchangeRate}`)
+    logger.info(` - Open date:                               ${this.config.instance.openDate}`)
+    logger.info(` - Reserve ratio:                           ${this.config.instance.reserveRatio}`)
+    logger.info(` - Batch blocks:                            ${this.config.instance.batchBlocks}`)
+    logger.info(` - Slippage:                                ${this.config.instance.slippage}`)
+  }
+
+  async _changeBondedTokenController(bondedTokenAddress) {
+    const MiniMeToken = await this.environment.getArtifact('MiniMeToken', '@aragon/minime')
+    const miniMeBondedToken = await MiniMeToken.at(bondedTokenAddress)
+    const currentController = await miniMeBondedToken.controller()
+    if (currentController == this.presaleTemplate.address) {
+      logger.warn(`Bonded token controller is already owned by template`)
+      return
+    }
+    const sender = await this.environment.getSender()
+    if (currentController != sender) {
+      throw `Bonded token controller is already owned by ${currentController}`
+    }
+    logger.info(`Changing bonded token controller to template`)
+    await miniMeBondedToken.changeController(this.presaleTemplate.address);
+  }
+
+  // TODO: remove once BalanceRedirectPresale is in aragen
+  async _publishBalanceRedirectPresale() {
```

Comment:
Can we move all these presale publishing outside this deployer? We could check that in the command file and call the "publisher" to do it and then call the template deployer.

Reply:
I'd also remove the TODOs and open an issue instead

Reply:
We'll remove it completely soon, but I'll move it away anyway.